s3buildcache/src/main/kotlin/androidx/build/gradle/s3buildcache/S3BuildCacheService.kt

```diff
@@ -69,6 +69,7 @@ class S3BuildCacheService(
     }
 
     override fun store(key: BuildCacheKey, writer: BuildCacheEntryWriter) {
+        if (writer.size == 0L) return // do not store empty entries into the cache
```

Comment:
This change is a copy of [this](https://github.com/androidx/gcp-gradle-build-cache/pull/36) one